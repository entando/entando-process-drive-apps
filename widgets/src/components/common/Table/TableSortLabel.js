import React from 'react';
import classNames from 'classnames';
import PropTypes from 'prop-types';
import MuiTableSortLabel from '@material-ui/core/TableSortLabel';
import withStyles from '@material-ui/styles/withStyles';

const styles = {
  ascActive: {
    '&:hover': {
      '@global svg': {
        transform: 'rotate(180deg)',
      },
    },
  },
  descActive: {
    '&:hover': {
      '@global svg': {
        transform: 'rotate(0deg)',
      },
    },
  },
  notActive: {
    '@global svg': {
      transform: 'rotate(0deg)',
    },
  },
};

const TableSortLabel = ({ direction, classes, active, className, children, onClick }) => {
  return (
    <MuiTableSortLabel
      className={classNames(classes[active ? `${direction}Active` : 'notActive'], className)}
      active={active}
      direction={direction}
      onClick={onClick}
    >
      {children}
    </MuiTableSortLabel>
  );
};

TableSortLabel.propTypes = {
  classes: PropTypes.shape({
    ascActive: PropTypes.string,
    descActive: PropTypes.string,
    notActive: PropTypes.string,
  }),
  direction: PropTypes.oneOf(['asc', 'desc']),
  active: PropTypes.bool.isRequired,
  className: PropTypes.string,
  children: PropTypes.string.isRequired,
  onClick: PropTypes.func.isRequired,
};

TableSortLabel.defaultProps = {
  classes: {},
  direction: 'asc',
  className: '',
};

export default withStyles(styles)(TableSortLabel);
