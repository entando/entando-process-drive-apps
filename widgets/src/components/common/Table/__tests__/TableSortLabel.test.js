import React from 'react';
import { fireEvent, render } from '@testing-library/react';

import TableSortLabel from 'components/common/Table/TableSortLabel';

describe('<TableSortLabel />', () => {
  it('renders snapshot correctly', () => {
    const onClick = jest.fn();
    const { container } = render(
      <TableSortLabel onClick={onClick} active={false}>
        Test
      </TableSortLabel>
    );

    expect(container).toMatchSnapshot();
  });

  it('renders snapshot correctly with desc direction and active', () => {
    const onClick = jest.fn();
    const { container, getByText } = render(
      <TableSortLabel onClick={onClick} direction="desc" active>
        Test
      </TableSortLabel>
    );

    fireEvent.click(getByText('Test'));

    expect(container).toMatchSnapshot();
    expect(onClick).toHaveBeenCalled();
  });
});
