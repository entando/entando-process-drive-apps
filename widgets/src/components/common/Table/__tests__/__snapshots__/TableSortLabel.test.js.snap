// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<TableSortLabel /> renders snapshot correctly 1`] = `
<div>
  <span
    aria-disabled="false"
    class="MuiButtonBase-root MuiTableSortLabel-root TableSortLabel-notActive-3"
    role="button"
    tabindex="0"
  >
    Test
    <svg
      aria-hidden="true"
      class="MuiSvgIcon-root MuiTableSortLabel-icon MuiTableSortLabel-iconDirectionAsc"
      focusable="false"
      role="presentation"
      viewBox="0 0 24 24"
    >
      <path
        d="M20 12l-1.41-1.41L13 16.17V4h-2v12.17l-5.58-5.59L4 12l8 8 8-8z"
      />
    </svg>
  </span>
</div>
`;

exports[`<TableSortLabel /> renders snapshot correctly with desc direction and active 1`] = `
<div>
  <span
    aria-disabled="false"
    class="MuiButtonBase-root MuiTableSortLabel-root TableSortLabel-descActive-22 MuiTableSortLabel-active"
    role="button"
    tabindex="0"
  >
    Test
    <svg
      aria-hidden="true"
      class="MuiSvgIcon-root MuiTableSortLabel-icon MuiTableSortLabel-iconDirectionDesc"
      focusable="false"
      role="presentation"
      viewBox="0 0 24 24"
    >
      <path
        d="M20 12l-1.41-1.41L13 16.17V4h-2v12.17l-5.58-5.59L4 12l8 8 8-8z"
      />
    </svg>
  </span>
</div>
`;
